api/app/controllers/mno_enterprise/jpi/v1/app_reviews_controller.rb

```diff
@@ -1,34 +1,67 @@
 module MnoEnterprise
   class Jpi::V1::AppReviewsController < Jpi::V1::BaseResourceController
+    before_action :ensure_app_exists
+
     # GET /mnoe/jpi/v1/marketplace/:id/app_reviews
     def index
-      @app_reviews = MnoEnterprise::AppReview.approved.where(reviewable_id: params[:id])
-      @app_reviews = @app_reviews.limit(params[:limit]) if params[:limit]
-      @app_reviews = @app_reviews.skip(params[:offset]) if params[:offset]
-      @app_reviews = @app_reviews.order_by(params[:order_by]) if params[:order_by]
-      @app_reviews = @app_reviews.where(params[:where]) if params[:where]
-      @app_reviews = @app_reviews.all.fetch
-      response.headers['X-Total-Count'] = @app_reviews.metadata[:pagination][:count]
+      index! do |chain|
```

Comment:
here I would keep the filtering and call the scope method doing all these operations